br/pkg/restore/client.go

```diff
@@ -2340,15 +2354,70 @@ func (rc *Client) CleanUpKVFiles(
 	return rc.fileImporter.ClearFiles(ctx, rc.pdClient, "v1")
 }
 
+func (rc *Client) initSchemasMap(
+	ctx context.Context,
+	clusterID uint64,
+	restoreTS uint64,
+) ([]*backuppb.PitrDBMap, error) {
+	filename := metautil.PitrIDMapsFilename(clusterID, restoreTS)
+	exist, err := rc.storage.FileExists(ctx, filename)
+	if err != nil {
+		return nil, errors.Annotatef(err, "failed to check filename:%s ", filename)
+	} else if !exist {
+		log.Info("pitr id maps isn't existed", zap.String("file", filename))
+		return nil, nil
+	}
+
+	metaData, err := rc.storage.ReadFile(ctx, filename)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	backupMeta := &backuppb.BackupMeta{}
+	if err = backupMeta.Unmarshal(metaData); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return backupMeta.GetDbMaps(), nil
+}
+
 // InitSchemasReplaceForDDL gets schemas information Mapping from old schemas to new schemas.
 // It is used to rewrite meta kv-event.
 func (rc *Client) InitSchemasReplaceForDDL(
-	dbMaps []*backuppb.PitrDBMap,
+	ctx context.Context,
+	tiflashRecorder *tiflashrec.TiFlashRecorder,
 	tables *map[int64]*metautil.Table,
 	tableFilter filter.Filter,
+	newTask bool,
+	hasFullRestore bool,
```

Comment:
if there are optional or default arguments, can we make them a config? So when are going to add more arguments in the future, we just need to add new fields in the config, and no need to change every call site of this function.

```go
type InitSchemaConfig struct {
  IsNewTask bool
  HasFullRestore bool
  TiFlashRecorder *tiflashrec.TiFlashRecorder
}
```